concourse/scripts/bosh_ensure_uploaded.sh

```diff
@@ -0,0 +1,29 @@
+#! /bin/sh
+set -e
+
+script_dir=$(cd "$(dirname "$0")" && pwd)
+
+type="${TYPE:-$1}"
+name="${NAME:-$2}"
+#shellcheck disable=SC2153
+version="${VERSION:=$3}"
+#shellcheck disable=SC2153
```

Comment:
We should only disable checks in exceptional cases and describe the reason why in the commit message.

From reading [SC2153](https://github.com/koalaman/shellcheck/wiki/SC2153) am I right in thinking that we've disabled this because the variable names are too alike? In which case, I think it's probably right to complain, since it's confusing to have upper and lowercased variables of the same name.

It looks like we only use this script in the pipeline with environment variables, not by passing arguments. In which case it'd be preferable to use the variables directly, remove the reassignments/defaults, and we could probably replace the validation with `set -u` to exit if any are not set.